underfs/swift/src/main/java/alluxio/underfs/swift/SwiftUnderFileSystemFactory.java

```diff
@@ -101,6 +101,12 @@ private boolean addAndCheckSwiftCredentials() {
       Configuration.set(passwordKeyConf, System.getProperty(passwordKeyConf));
     }
 
+    String swiftSimulationConf = Constants.SWIFT_SIMULATION;
+    if (System.getProperty(swiftSimulationConf) != null
+        && !Configuration.containsKey(swiftSimulationConf)) {
```

Comment:
Sure, that sounds good.
But actually my point here is that, we probably need the following?
```
    if (System.getProperty(swiftSimulationConf) != null
       && (!Configuration.containsKey(swiftSimulationConf)
              || Configuration.get(swiftSimulationConf) == null)) {
```